kinds/dashboard/dashboard_kind.cue

```diff
@@ -186,12 +187,160 @@ lineage: schemas: [{
+			options: [...#VariableOption]
+			query: string
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Option to be selected in a variable.
+		#VariableOption: {
+			selected: bool
+			text:     string | [...string]
+			value:    string | [...string]
+			isNone?:  bool
+		} @cuetsy(kind="interface")
+
+		// Variable which value is set by a free text with an optional default value.
+		#TextBoxVariableModel: {
+			#VariableWithOptions
+			type: "textbox"
+			// This can be null
+			originalQuery: string
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variable which value cannot be changed
+		#ConstantVariableModel: {
+			#VariableWithOptions
+			type: "constant"
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variables to represents time spans such as 1m, 1h, 1d.
+		#IntervalVariableModel: {
+			#VariableWithOptions
+			type: "interval"
+			// This option allows you to specify how many times the current time range should be divided to calculate the current `auto` time span.
+			// Configuring `auto_min` and or `auto_count` will setup the auto interval to use the min or count option.
+			auto: bool
+			// The minimum threshold below which the step count intervals will not divide the time. 
+			// For example, if the minimum interval is set to `2m`, and the value is `30m` then Grafana would group the data into 15 two-minute increments.
+			auto_min: string
+			// Select the number of times the current time range will be divided to calculate the value, similar to the Max data points query option. 
+			// For example, if the current visible time range is 30 minutes, then the auto interval groups the data into 30 one-minute increments.
+			auto_count: int32
+			refresh:    #VariableRefresh
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variable which value is selected from a list of options and it support the selection of multiple values.
+		#VariableWithMultiSupport: {
+			#VariableWithOptions
+			multi:      bool
+			includeAll: bool
+			allValue?:  string
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Query-generated list of values such as metric names, server names, sensor IDs, data centers, and so on.
+		#QueryVariableModel: {
+			#VariableWithMultiSupport
+			type:        "query"
+			datasource?: #DataSourceRef
+			definition:  string
+			sort:        #VariableSort
+			queryValue?: string
+			query:       string | {...}
+			regex:       string
+			refresh:     #VariableRefresh
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Variable which options are different data sources from the same type.
+		#DataSourceVariableModel: {
+			#VariableWithMultiSupport
+			type:    "datasource"
+			regex:   string
+			refresh: #VariableRefresh
+		} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+		// Define the variable options manually using a comma-separated list.
+		#CustomVariableModel: {
+			#VariableWithMultiSupport
+			type: "custom"
```

Comment:
Its overriding `type` value from `#BaseVariableModel`. This value is defined as `#VariableType`, so the correct way to override should be `#VariableType & "custom"`. It should be update for all other models.